### PR TITLE
feat!: Remove dependency on itwinui-icons

### LINKED ISSUE #912: v2: Remove CJS icon imports and instead just inline the svgs

### PR DESCRIPTION
**Pull request by mayank99**
Removed `@itwin/itwinui-icons-react` and moved the icons that are used by our components into `src/utils/icons`.

First step towards #912. In a separate PR, I will change illustrations to use dynamic imports.

### CODE REVIEW

**Review by gretanausedaite — APPROVED** [review]
I have no comments on code. Looks good.

File: packages/iTwinUI-react/src/core/ColorPicker/ColorPalette.test.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { ColorPalette } from './ColorPalette';
 import { ColorPicker } from './ColorPicker';
 import { IconButton } from '../Buttons';
 import { ColorValue } from '../utils';
-import { SvgAdd } from '@itwin/itwinui-icons-react';
+import { SvgMore } from '../utils';
```
Comment (adamhock) [inline packages/iTwinUI-react/src/core/ColorPicker/ColorPalette.test.tsx:11]: This import can be combined with the `ColorValue` import

File: packages/iTwinUI-react/src/core/ExpandableBlock/ExpandableBlock.test.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import { fireEvent, render } from '@testing-library/react';
 import React from 'react';
 import { StatusIconMap } from '../utils';
-import { SvgPlaceholder } from '@itwin/itwinui-icons-react';
+import { SvgMore as SvgPlaceholder } from '../utils';
```
Comment (adamhock) [inline packages/iTwinUI-react/src/core/ExpandableBlock/ExpandableBlock.test.tsx:8]: This import can be combined with the `StatusIconMap` import

File: packages/iTwinUI-react/src/core/Header/HeaderDropdownButton.tsx [hunk 3]
```diff
@@ -5,11 +5,9 @@
 import cx from 'classnames';
 import React from 'react';
 import { DropdownMenu } from '../DropdownMenu';
-import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
-import SvgCaretUpSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretUpSmall';
 import { PolymorphicForwardRefComponent, useTheme } from '../utils';
 import { DropdownButtonProps } from '../Buttons';
-import { useMergedRefs } from '../utils';
+import { useMergedRefs, SvgCaretDownSmall, SvgCaretUpSmall } from '../utils';
```
Comment (adamhock) [inline packages/iTwinUI-react/src/core/Header/HeaderDropdownButton.tsx:10]: These imports can be combined with the `PolymorphicForwardRefComponent` and `useTheme` imports. I'm not sure why they were separated in the first place

**Review by adamhock — COMMENTED** [review]
Aside from combining some imports, the code looks good to me. I'll approve it once the comments are resolved

File: packages/iTwinUI-react/src/core/ColorPicker/ColorPalette.test.tsx (hunk 1 above)
Comment (mayank99) [reply packages/iTwinUI-react/src/core/ColorPicker/ColorPalette.test.tsx]: Done

File: packages/iTwinUI-react/src/core/ExpandableBlock/ExpandableBlock.test.tsx (hunk 2 above)
Comment (mayank99) [reply packages/iTwinUI-react/src/core/ExpandableBlock/ExpandableBlock.test.tsx]: Done

File: packages/iTwinUI-react/src/core/Header/HeaderDropdownButton.tsx (hunk 3 above)
Comment (mayank99) [reply packages/iTwinUI-react/src/core/Header/HeaderDropdownButton.tsx]: Done

**Review by adamhock — APPROVED** [review]

**Comment by rohan-kadkol** [comment]
To confirm, do we need to update the style guide regarding importing icons in components (`src/core`)?

**Comment by mayank99** [comment]
> To confirm, do we need to update the style guide regarding importing icons in components (`src/core`)?

Yeah we should.

**Pull request merged**